### PR TITLE
Add XCommon CMake support in examples and tests

### PR DESCRIPTION
**Pull request by danielpieczko**
- examples converted to build using XCommon CMake; rename directory `example` -> `examples`
- build/run instructions in example READMEs updated
- tests converted to build using XCommon CMake; rename directory `test` -> `tests`
- lib_unity is used instead of the upstream Unity that was previously fetched by CMake
- the changes to the tests can be hard to see in the diff because of the renamed directory
- the CMakeLists.txt are a bit complicated with different compiler flags based on the compiler (xcc, gcc, msvc) but I've based this on the old flags and it's duplicated in each test app
- one test app also has some Python script auto-generation that was commented out: I've recreated and tested this with XCommon CMake, but it remains commented out
- native builds on Windows can't be built without compiler/linker warnings due to a [limitation in XCommon CMake](https://github.com/xmos/xcommon_cmake/issues/189)
- legacy CMake test added to the legacy XCommon test
- use preferred XN target `XK-EVK-XU316` instead of `XCORE-AI-EXPLORER` throughout
- update versions to v2.3.0 ready for release

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -129,23 +118,26 @@ pipeline {
...
+                        bat 'cmake -B build_x86 -G Ninja -D BUILD_NATIVE=TRUE'
+                        bat 'ninja -C build_x86'
+                      }
+                      dir('tests/legacy_build') {
+                        // legacy XCommon
                         bat 'xmake --jobs 4'
+                        // legacy CMake
+                        bat "cmake -B build --toolchain=${WORKSPACE}/lib_xcore_math/etc/xmos_cmake_toolchain/xs3a.cmake -G Ninja"
+                        bat 'ninja -C build'
                       }
```
Comment (ed-xmos) [inline Jenkinsfile:141]: I think ninja is multi-process by default? If so all good.

File: tests/filter_tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                                   )
+endif()
+
+if(PLATFORM_ID STREQUAL Linux OR PLATFORM_ID STREQUAL Darwin)
+    list(APPEND APP_COMPILER_FLAGS -lm)
+endif()
+
+XMOS_REGISTER_APP()
+
+#########
```
Comment (ed-xmos) [inline tests/filter_tests/CMakeLists.txt:56]: Any reason this is left but commented?

**Review by ed-xmos — APPROVED** [review]
Actually that wasn't too bad - mostly renames and the same cmake pattern over and over. My left mouse button isn't too happy tho.

File: tests/filter_tests/CMakeLists.txt (hunk 2 above)
Comment (danielpieczko) [reply tests/filter_tests/CMakeLists.txt:56]: Just that it was commented in the original, and I wrote an equivalent, so I've left it for whatever reason it was there for.

**Review by shuchitak — APPROVED** [review]
Looks good :)

**Pull request merged**